### PR TITLE
[restclient-cpp] Fix portfile and update version.

### PR DESCRIPTION
**Pull request by et-nik**
This PR to update restclient library and fix portfile. Previously, the portfile had a command to delete the all compiled files (`restclient-cpp.lib`, `restclient-cppd.lib`), it is mistake.

I update library version from 0.5.1 tag to last commit. CMakeLists patches are no longer required in the new library version.
Library is updated and bugfixes, but the developer did not add new tags. Some people want tag updating (https://github.com/mrtazz/restclient-cpp/issues/145).

### CODE REVIEW

File: ports/restclient-cpp/portfile.cmake [hunk 1]
```diff
@@ -3,27 +3,28 @@ include(vcpkg_common_functions)
 vcpkg_from_github(
```
Comment (NancyLi1013) [inline ports/restclient-cpp/portfile.cmake]: `include(vcpkg_common_functions)` is no longer needed.
Could you remove this line?

File: ports/restclient-cpp/portfile.cmake [hunk 2]
```diff
@@ -3,27 +3,28 @@ include(vcpkg_common_functions)
...
 vcpkg_install_cmake()
 
 vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/restclient-cpp)
 
 # Remove includes in debug
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug ${CURRENT_PACKAGES_DIR}/lib ${CURRENT_PACKAGES_DIR}/bin)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
 
 # Handle copyright
 file(COPY ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/restclient-cpp)
```
Comment (NancyLi1013) [inline ports/restclient-cpp/portfile.cmake]: Could you update 
```
file(COPY ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/restclient-cpp)
file(RENAME ${CURRENT_PACKAGES_DIR}/share/restclient-cpp/LICENSE ${CURRENT_PACKAGES_DIR}/share/restclient-cpp/copyright)
```
as
`file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)`?

File: ports/restclient-cpp/portfile.cmake [hunk 3]
```diff
@@ -3,27 +3,28 @@ include(vcpkg_common_functions)
...
-    REF 0.5.1
-    SHA512 d5e17a984af44f863bc7cdc7307c2b06cae9252f86c6c6c2377cdb317f61b6419d8e9aedc5e5ccdb08fd1ee13848ec3b9ef8067a8d26dcf438a5c8793b5a2ce3
+    REF 6336cae5275c9aeddf99f13c49e8f9320f7ca4bc
+    SHA512 fbc638f82db8d0d101f4239152be67eb460659f3a7204db9523d3b5740b007b8fe434f78bf5d9f8059901aa639ba96c15b28b1e2fe5f09471accf64c2d577684 
     HEAD_REF master
-    PATCHES
-        0001_fix_cmake_linking.patch
 )
 
+set(COMPILE_TYPE STATIC)
```
Comment (NancyLi1013) [inline ports/restclient-cpp/portfile.cmake]: Why do you set `COMPILE_TYPE `as `STATIC ` here ?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
Hi @et-nik Thanks for this PR.
Could you also bump the version in `CONTROL `file?

File: ports/restclient-cpp/portfile.cmake (hunk 1 above)
Comment (et-nik) [reply ports/restclient-cpp/portfile.cmake]: Removed

File: ports/restclient-cpp/portfile.cmake (hunk 3 above)
Comment (et-nik) [reply ports/restclient-cpp/portfile.cmake]: If `COMPILE_TYPE` is `STATIC` set, it will be compiled lib file.
If `COMPILE_TYPE` not set, it will be compiled dll file.

It is not CRT parameter. And does not affect CRT compile parameters.

See https://github.com/mrtazz/restclient-cpp/blob/master/CMakeLists.txt#L46

I moved from here to cmake options `-DCOMPILE_TYPE=STATIC`

**Comment by et-nik** [comment]
Version changed in `CONTROL` file.

`vcpkg-osx-PR-test` still failed. I think this fail is not related port and package.

**Comment by et-nik** [comment]
Hm. Now _vcpkg-windows-PR-test_ failed

**Comment by et-nik** [comment]
@NancyLi1013 could you run tests again?

**Comment by NancyLi1013** [comment]
@et-nik thanks for your remind.
The PR failed on windows platform due to` ppconsul`.
PR #9752 will fix the problem.
So let's wait for PR #9752 merged, then I will re run this PR again.

**Review by NancyLi1013 — APPROVED** [review]

File: ports/restclient-cpp/portfile.cmake [hunk 4]
```diff
@@ -16,18 +12,18 @@ vcpkg_configure_cmake(
     OPTIONS
         -DCMAKE_DISABLE_FIND_PACKAGE_GTest=TRUE
         -DCMAKE_DISABLE_FIND_PACKAGE_jsoncpp=TRUE
+        -DCOMPILE_TYPE=STATIC
```
Comment (vicroms) [inline ports/restclient-cpp/portfile.cmake]: Why is this necessary? Does it affect dynamic library builds?

Comment (et-nik) [reply ports/restclient-cpp/portfile.cmake]: See https://github.com/microsoft/vcpkg/pull/9717#discussion_r367821525

This is not the same `static` triplets vcpkg thing.

Default `COMPILE_TYPE` is SHARED. It is mean than only .dll/.so library will be compile.
I agree that this is a little confusing. I can add CMakeLists.txt patch instead of this option.

Comment (et-nik) [reply ports/restclient-cpp/portfile.cmake]: @vicroms Restclient `CmakeList.txt` is made in such a way that it is compile to either lib or dll.
We can't way to build both (lib and dll).
Without `-DCOMPILE_TYPE=STATIC`, only dynamic library will be compiled.
I think it's better to build static libraries, or both. Patch `CmakeList.txt` is needed if we want build static and dynamic libraries.

File: ports/restclient-cpp/CONTROL [hunk 5]
```diff
@@ -1,4 +1,4 @@
 Source: restclient-cpp
-Version: 0.5.1-2
+Version: 0.5.1-3
 Build-Depends: curl
 Description: Binn is a binary data serialization format designed to be compact, fast and easy to use.
```
Comment (maksqwe) [inline ports/restclient-cpp/CONTROL]: I think it is incorrect description, could you fix it?

Comment (et-nik) [reply ports/restclient-cpp/CONTROL]: Fixed. Also I added Homepage info

**Review by maksqwe — APPROVED** [review]

File: ports/restclient-cpp/portfile.cmake (hunk 4 above)
Comment (strega-nil) [reply ports/restclient-cpp/portfile.cmake]: @ras0219-msft What's the opinion of vcpkg here? ^

Comment (dan-shaw) [reply ports/restclient-cpp/portfile.cmake]: We should use `VCPKG_LIBRARY_LINKAGE`
See https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#choose-either-static-or-shared-binaries

Comment (et-nik) [reply ports/restclient-cpp/portfile.cmake]: Okay. Thanks.
I'll fix. Also I'll bump restclient version. 0.5.2 release is available

**Comment by NancyLi1013** [comment]
@et-nik 
Could you please address the review suggestion?

**Comment by et-nik** [comment]
Hello. I'll update this PR today.

**Comment by NancyLi1013** [comment]
It seems that the fix is not correct. Please refer to this [doc](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#choose-either-static-or-shared-binaries) to try it again.

**Comment by et-nik** [comment]
Done 

* Bumped version to last tag 0.5.2
* Removed old patches
* Fixed descriptions

File: ports/restclient-cpp/CONTROL [hunk 6]
```diff
@@ -1,4 +1,5 @@
 Source: restclient-cpp
-Version: 0.5.1-3
+Version: 0.5.2-1
```
Comment (NancyLi1013) [inline ports/restclient-cpp/CONTROL:2]: The version should be `0.5.2` instead of` 0.5.2-1`.
Could you please update it again?

**Comment by et-nik** [comment]
@NancyLi1013 Done

Hm. I see that all checks are succeeded https://dev.azure.com/vcpkg/public/_build/results?buildId=36451&view=results

But here check statuses are still "Waiting for..."

**Comment by strega-nil** [comment]
Cool, looks good to me! Thanks @et-nik :)
